src/main/java/svc/controllers/CourtController.java

```diff
@@ -30,11 +29,8 @@ CourtsDTO GetCourts() {
 	
 	@ResponseBody
 	@RequestMapping(method = RequestMethod.GET, value="/{id}")
-	Court GetCourt(@PathVariable("id") Long id) {
-		if (id == null) {
-			LogSystem.LogEvent("Null id passed to controller");
-		}
-		
-		return courtManager.GetCourtById(id);
+	Court GetCourt(@PathVariable("id") Long id) throws NotFoundException {

```

Comment:
Nothing throws the exception here...unless I don't understand the framework, I don't think "throws NotFoundException" will do anything if court is null....instead, remove that and check:

```if(court == null) {
  throw new NotFoundException();
}```